modules/newspassidBidAdapter.js

```diff
@@ -0,0 +1,639 @@
+              adserverTargeting['np_' + bidderName + '_dealid'] = String(allBidsForThisBidid[bidderName].dealid);
+            }
+          });
+        } else {
+          logInfo(`newspassid.enhancedAdserverTargeting is set to false, no per-bid keys will be sent to adserver.`);
+        }
+        let {seat: winningSeat, bid: winningBid} = this.getWinnerForRequestBid(thisBid.bidId, serverResponse.seatbid);
+        adserverTargeting['np_auc_id'] = String(request.bidderRequest.auctionId);
+        adserverTargeting['np_winner'] = String(winningSeat);
+        adserverTargeting['np_bid'] = 'true';
+        if (enhancedAdserverTargeting) {
+          adserverTargeting['np_imp_id'] = String(winningBid.impid);
+          adserverTargeting['np_pb_r'] = getRoundedBid(winningBid.price, bidType);
+          adserverTargeting['np_adId'] = String(winningBid.adId);
+          adserverTargeting['np_size'] = `${winningBid.width}x${winningBid.height}`;
+        }
+        if (useWhitelistAdserverKeys) { // delete any un-whitelisted keys
+          logInfo('Going to filter out adserver targeting keys not in the whitelist: ', whitelistAdserverKeys);
+          Object.keys(adserverTargeting).forEach(function(key) { if (whitelistAdserverKeys.indexOf(key) === -1) { delete adserverTargeting[key]; } });
+        }
+        thisBid.adserverTargeting = adserverTargeting;
+        arrAllBids.push(thisBid);
+      }
+    }
+    let endTime = new Date().getTime();
+    logInfo(`interpretResponse going to return at time ${endTime} (took ${endTime - startTime}ms) Time from buildRequests Start -> interpretRequests End = ${endTime - this.propertyBag.buildRequestsStart}ms`, arrAllBids);
+    return arrAllBids;
+  },
+  removeSingleBidderMultipleBids(seatbid) {
+    var ret = [];
+    for (let i = 0; i < seatbid.length; i++) {
+      let sb = seatbid[i];
+      var retSeatbid = {'seat': sb.seat, 'bid': []};
+      var bidIds = [];
+      for (let j = 0; j < sb.bid.length; j++) {
+        var candidate = sb.bid[j];
+        if (contains(bidIds, candidate.impid)) {
+          continue; // we've already fully assessed this impid, found the highest bid from this seat for it
+        }
+        bidIds.push(candidate.impid);
+        for (let k = j + 1; k < sb.bid.length; k++) {
+          if (sb.bid[k].impid === candidate.impid && sb.bid[k].price > candidate.price) {
+            candidate = sb.bid[k];
+          }
+        }
+        retSeatbid.bid.push(candidate);
+      }
+      ret.push(retSeatbid);
+    }
+    return ret;
+  },
+  getUserSyncs(optionsType, serverResponse, gdprConsent, usPrivacy) {
+    logInfo('getUserSyncs optionsType', optionsType, 'serverResponse', serverResponse, 'usPrivacy', usPrivacy, 'cookieSyncBag', this.cookieSyncBag);
+    if (!serverResponse || serverResponse.length === 0) {
+      return [];
+    }
+    if (optionsType.iframeEnabled) {
+      var arrQueryString = [];
+      if (getRefererInfo().page.match(/pbjs_debug=true/)) {
```

Comment:
why not like this https://github.com/prebid/Prebid.js/blob/6f4fb5b78a4bc31a1f84b59ec5b98d4e3bcf0179/modules/adWMGBidAdapter.js#L29

Reply:
have updated